cdc/sinkv2/ddlsink/txn/mysql_ddl_sink.go

```diff
@@ -0,0 +1,200 @@
+	if err != nil {
+		return nil, err
+	}
+
+	m := &mysqlDDLSink{
+		id:         changefeedID,
+		db:         db,
+		statistics: metrics.NewStatistics(ctx, sink.TxnSink),
+	}
+
+	log.Info("MySQL DDL sink is created",
+		zap.String("namespace", m.id.Namespace),
+		zap.String("changefeed", m.id.ID))
+	return m, nil
+}
+
+func (m *mysqlDDLSink) WriteDDLEvent(ctx context.Context, ddl *model.DDLEvent) error {
+	m.statistics.AddDDLCount()
+	err := m.execDDLWithMaxRetries(ctx, ddl)
+	return errors.Trace(err)
+}
+
+func (m *mysqlDDLSink) execDDLWithMaxRetries(ctx context.Context, ddl *model.DDLEvent) error {
+	return retry.Do(ctx, func() error {
+		err := m.execDDL(ctx, ddl)
+		if err != nil {
+			if errorutil.IsIgnorableMySQLDDLError(err) {
+				log.Info("Execute DDL failed, but error can be ignored",
+					zap.Uint64("startTs", ddl.StartTs), zap.String("ddl", ddl.Query),
+					zap.String("namespace", m.id.Namespace),
+					zap.String("changefeed", m.id.ID),
+					zap.Error(err))
+				// If the error is ignorable, we will direly ignore the error.
+				return nil
+			}
+			log.Warn("Execute DDL with error, retry later",
+				zap.Uint64("startTs", ddl.StartTs), zap.String("ddl", ddl.Query),
+				zap.String("namespace", m.id.Namespace),
+				zap.String("changefeed", m.id.ID),
+				zap.Error(err))
+			return err
+		}
+		return nil
+	}, retry.WithBackoffBaseDelay(pmysql.BackoffBaseDelay.Milliseconds()),
+		retry.WithBackoffMaxDelay(pmysql.BackoffMaxDelay.Milliseconds()),
+		retry.WithMaxTries(defaultDDLMaxRetry),
+		retry.WithIsRetryableErr(cerror.IsRetryableError))
+}
+
+func (m *mysqlDDLSink) execDDL(ctx context.Context, ddl *model.DDLEvent) error {
+	shouldSwitchDB := needSwitchDB(ddl)
+
+	failpoint.Inject("MySQLSinkExecDDLDelay", func() {
+		select {
+		case <-ctx.Done():
+			failpoint.Return(ctx.Err())
+		case <-time.After(time.Hour):
+		}
+		failpoint.Return(nil)
+	})
+
+	log.Info("Start exec DDL", zap.Any("DDL", ddl), zap.String("namespace", m.id.Namespace),
+		zap.String("changefeed", m.id.ID))
+	err := m.statistics.RecordDDLExecution(func() error {
+		tx, err := m.db.BeginTx(ctx, nil)
+		if err != nil {
+			return err
+		}
+
+		if shouldSwitchDB {
+			_, err = tx.ExecContext(ctx, "USE "+quotes.QuoteName(ddl.TableInfo.Schema)+";")
+			if err != nil {
+				if rbErr := tx.Rollback(); rbErr != nil {
+					log.Error("Failed to rollback", zap.String("namespace", m.id.Namespace),
+						zap.String("changefeed", m.id.ID), zap.Error(err))
+				}
+				return err
+			}
+		}
+
+		if _, err = tx.ExecContext(ctx, ddl.Query); err != nil {
+			if rbErr := tx.Rollback(); rbErr != nil {
+				log.Error("Failed to rollback", zap.String("sql", ddl.Query),
+					zap.String("namespace", m.id.Namespace),
+					zap.String("changefeed", m.id.ID), zap.Error(err))
+			}
+			return err
+		}
+
+		return tx.Commit()
+	})
+	if err != nil {
+		return cerror.WrapError(cerror.ErrMySQLTxnError, err)
+	}
+
+	log.Info("Exec DDL succeeded", zap.String("sql", ddl.Query),
```

Comment:
if the time used exceed a value , it's better to record the information for investigation later

Reply:
Added. Thanks!